test/graphreader.cc

```diff
@@ -146,33 +156,33 @@ TEST(SimpleCache, Clear) {
   SimpleTileCache cache(400);
 
   GraphId id1(100, 2, 0);
-  TestGraphTile tile1(id1, 123);
-  const GraphTile* inserted1 = cache.Put(id1, tile1, 123);
-  CheckGraphTile(inserted1, id1, 123);
+  const auto tile1 = cache.Put(id1, TestGraphTile(id1, 123), 123);
+  EXPECT_EQ(cache.Get(id1), tile1);
+  CheckGraphTile(tile1, id1, 123);
 
   EXPECT_FALSE(cache.OverCommitted());
 
   GraphId id2(300, 1, 0);
-  TestGraphTile tile2(id2, 200);
-  const GraphTile* inserted2 = cache.Put(id2, tile2, 200);
-  CheckGraphTile(inserted2, id2, 200);
+  const auto tile2 = cache.Put(id2, TestGraphTile(id2, 200), 200);
+  EXPECT_EQ(cache.Get(id2), tile2);
+  CheckGraphTile(tile2, id2, 200);
 
   EXPECT_FALSE(cache.OverCommitted());
 
   GraphId id3(1000, 0, 0);
-  TestGraphTile tile3(id3, 500);
-  const GraphTile* inserted3 = cache.Put(id3, tile3, 500);
-  CheckGraphTile(inserted3, id3, 500);
+  const auto tile3 = cache.Put(id3, TestGraphTile(id3, 500), 500);
+  EXPECT_EQ(cache.Get(id3), tile3);
+  CheckGraphTile(tile3, id3, 500);
 
   EXPECT_TRUE(cache.OverCommitted());
 
   // Check if inserted values are correct
 
-  const auto* returned2 = cache.Get({300, 1, 0});
+  const auto returned2 = cache.Get({300, 1, 0});
   CheckGraphTile(returned2, id2, 200);
-  const auto* returned1 = cache.Get({100, 2, 0});
+  const auto returned1 = cache.Get({100, 2, 0});
   CheckGraphTile(returned1, id1, 123);
-  const auto* returned3 = cache.Get({1000, 0, 0});
+  const auto returned3 = cache.Get({1000, 0, 0});
```

Comment:
these are pointers, i know type deduction will include the pointer in `auto` but its less illustrative of what is going on to remove the pointer. can we revert these please

Reply:
seems this whole file has similar changes it would be nice to revert all of them

Reply:
Get returns boost::intrusive_ptr<const baldr::GraphTile>, so this is not a raw pointer.

Reply:
not in this pr it doesn :smile:

Reply:
Ok, will add *